presto-native-execution/presto_cpp/main/common/PrometheusStatsReporter.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/dynamic.h>
+#include <fstream>
+#include <iostream>
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/common/Counters.h"
+#include "velox/common/base/StatsReporter.h"
+#include "velox/velox/common/base/Exceptions.h"
+
+namespace ahana {
+
+class PrometheusStatsReporter : public facebook::velox::BaseStatsReporter {
+ public:
+  PrometheusStatsReporter(
+      const std::string cluster = "",
+      const std::string worker = "") {
+    if (cluster.empty()) {
+      auto nodeConfig = facebook::presto::NodeConfig::instance();
+      cluster_ = nodeConfig->nodeEnvironment();
+    } else {
+      cluster_ = cluster;
+    }
+    workerPod_ = worker.empty() ? std::getenv("HOSTNAME") : worker;
+  }
+
+  struct Gauge {
+    uint64_t timestamp; // Epoch timestamp.
+    size_t value; // Metric value.
+  };
+
+  /// Register a stat of the given stat type.
+  /// @param key The key to identify the stat.
+  /// @param statType How the stat is aggregated.
+  void registerMetricExportType(
+      const char* key,
+      facebook::velox::StatType statType) const override;
+
+  void registerMetricExportType(
+      folly::StringPiece key,
+      facebook::velox::StatType statType) const override;
+
+  void registerHistogramMetricExportType(
+      const char* /*key*/,
+      int64_t /* bucketWidth */,
+      int64_t /* min */,
+      int64_t /* max */,
+      const std::vector<int32_t>& /* pcts */) const override {}
+
+  void registerHistogramMetricExportType(
+      folly::StringPiece /* key */,
+      int64_t /* bucketWidth */,
+      int64_t /* min */,
+      int64_t /* max */,
+      const std::vector<int32_t>& /* pcts */) const override {}
+
+  void addMetricValue(const std::string& key, size_t value = 1)
+      const override{};
```

Comment:
Why is this an empty method ? We should do the same logic as the other addMetricValue methods.